src/lib/statistics/chunk_statistics/histograms/equal_width_histogram.cpp

```diff
@@ -0,0 +1,281 @@
+  return _bin_data.bin_heights.size();
+}
+
+template <typename T>
+HistogramCountType EqualWidthHistogram<T>::_bin_height(const BinID index) const {
+  DebugAssert(index < _bin_data.bin_heights.size(), "Index is not a valid bin.");
+  return _bin_data.bin_heights[index];
+}
+
+template <typename T>
+HistogramCountType EqualWidthHistogram<T>::total_count() const {
+  return std::accumulate(_bin_data.bin_heights.cbegin(), _bin_data.bin_heights.cend(), HistogramCountType{0});
+}
+
+template <typename T>
+HistogramCountType EqualWidthHistogram<T>::total_distinct_count() const {
+  return std::accumulate(_bin_data.bin_distinct_counts.cbegin(), _bin_data.bin_distinct_counts.cend(),
+                         HistogramCountType{0});
+}
+
+template <typename T>
+HistogramCountType EqualWidthHistogram<T>::_bin_distinct_count(const BinID index) const {
+  DebugAssert(index < _bin_data.bin_distinct_counts.size(), "Index is not a valid bin.");
+  return _bin_data.bin_distinct_counts[index];
+}
+
+template <typename T>
+typename AbstractHistogram<T>::HistogramWidthType EqualWidthHistogram<T>::_bin_width([
+    [maybe_unused]] const BinID index) const {
+  DebugAssert(index < bin_count(), "Index is not a valid bin.");
+
+  const auto base_width = this->_get_next_value(_bin_data.maximum - _bin_data.minimum) / bin_count();
+
+  if constexpr (std::is_integral_v<T>) {  // NOLINT
+    return base_width + (index < _bin_data.bin_count_with_larger_range ? 1 : 0);
+  }
+
+  return base_width;
+}
+
+template <>
+AbstractHistogram<std::string>::HistogramWidthType EqualWidthHistogram<std::string>::_bin_width(
+    const BinID index) const {
+  return AbstractHistogram<std::string>::_bin_width(index);
+}
+
+template <typename T>
+T EqualWidthHistogram<T>::_bin_minimum(const BinID index) const {
+  DebugAssert(index < bin_count(), "Index is not a valid bin.");
+
+  // If it's the first bin, return _minimum.
+  if (index == 0u) {
+    return _bin_data.minimum;
+  }
+
+  // Otherwise, return the next representable value of the previous bin's max.
+  return this->_get_next_value(_bin_maximum(index - 1));
+}
+
+template <typename T>
+T EqualWidthHistogram<T>::_bin_maximum(const BinID index) const {
+  DebugAssert(index < bin_count(), "Index is not a valid bin.");
+
+  // If it's the last bin, return max.
+  if (index == bin_count() - 1) {
+    return _bin_data.maximum;
+  }
+
+  // Calculate the lower edge of the bin right after index, assuming every bin has the same width.
+  // If there are no wider bins, take the previous value from the lower edge of the following bin to get the
+  // upper edge of this one.
+  // Otherwise, add the index to compensate one element for every bin preceding this bin.
+  // Add at most bin_count_with_larger_range - 1 because we already start adding from the next bin's lower edge.
+  if constexpr (std::is_same_v<T, std::string>) {
+    const auto num_min = this->_convert_string_to_number_representation(_bin_data.minimum);
```

Comment:
in _build_bins you called this "repr_min" didn't you? "num_min" could be read as "number of minima", whatever that is ;).